src/main/java/net/fabricmc/loom/configuration/ifaceinject/InterfaceInjectionProcessor.java

```diff
@@ -114,6 +118,11 @@ public void processJar(Path jar, Spec spec, ProcessorContext context) throws IOE
 
 		try (LazyCloseable<TinyRemapper> tinyRemapper = context.createRemapper(MappingsNamespace.INTERMEDIARY, MappingsNamespace.NAMED)) {
 			final List<InjectedInterface> remappedInjectedInterfaces = spec.injectedInterfaces().stream()
+					.filter(injectedInterface -> {
+						return context.includesClient() // The client jar depends on the server, so always apply all to it
+								|| context.isMerged() // Always apply all to merged jars
```

Comment:
Note to self, this line isnt needed as the merged jar always includes the client.